lib/ts/recipe/dashboard/api/userroles/permissions/addPermissions.ts

```diff
@@ -0,0 +1,38 @@
+import UserRoles from "../../../../userroles";
+import { APIInterface, APIOptions } from "../../../types";
+
+import STError from "../../../../../error";
+
+const addPermissions = async (
```

Comment:
this API also creates a role if it doesn't already exist. Is that taken into account? Maybe call this API `createNewRoleOrAddPermissions`, and reuse this API instead of having another API for createRole?